editor.planx.uk/src/@planx/components/SetFee/Editor.tsx

```diff
@@ -27,12 +43,141 @@ function SetFeeComponent(props: Props) {
   return (
     <form onSubmit={formik.handleSubmit} id="modal">
       <ModalSection>
-        <ModalSectionContent title="Passport field name">
-          <DataFieldAutocomplete
-            required
-            value={formik.values.fn}
-            onChange={(value) => formik.setFieldValue("fn", value)}
+        <ModalSectionContent
+          title="Set service fees"
+          Icon={ICONS[TYPES.SetFee]}
+        />
+      </ModalSection>
+      <ModalSection>
+        <ModalSectionContent title="Application fee VAT">
+          <InputRow>
+            <Switch
+              checked={formik.values.applyCalculatedVAT}
+              onChange={() =>
+                formik.setFieldValue(
+                  "applyCalculatedVAT",
+                  !formik.values.applyCalculatedVAT,
+                )
+              }
+              label="Apply VAT to the application fee"
+            />
+          </InputRow>
+          <Typography variant="body2">
+            If this is a discretionary service, apply 20% VAT to the application
+            fee. The incoming <strong>application.fee.calculated</strong> should
+            be exclusive of VAT and already reflect any exemptions, reductions
+            or increases.
+          </Typography>
+        </ModalSectionContent>
+      </ModalSection>
+      <ModalSection>
+        <ModalSectionContent title="Fast Track fee">
+          <Typography variant="body2" mb={2}>
+            If this service supports Fast Track journeys, specify the fee
+            amount. This fee plus 20% VAT will be added when{" "}
+            <strong>application.fastTrack</strong> applies.
+          </Typography>
+          <InputRow>
+            <InputRowLabel>£</InputRowLabel>
+            <Input
+              name="fastTrackFeeAmount"
+              placeholder="Amount (exclusive of VAT)"
+              value={formik.values.fastTrackFeeAmount}
```

Comment:
```suggestion
              value={formik.values.fastTrackFeeAmount || ""}
```

This is a frustrating React quirk we need to consider - we can't have `null` or `undefined` be the initial form value if it's later changed.

This means we need to populate with an empty string initially, then ensure we set falsy values back to what we want in `handleSetFees()`.

![image](https://github.com/user-attachments/assets/93ccf130-243c-45c6-adad-e52bfbc5bd84)

Reply:
`formik.values.fastTrackAmount` now required with default 0, so will never have `null` or `undefined` here which clears this up :+1: (note that 0 _displays_ as undefined in order to make placeholder text visible per later comment)